cache/compression/compression.go

```diff
@@ -1,21 +1,201 @@
 package compression
 
 import (
+	"archive/tar"
+	"bytes"
 	"errors"
 	"fmt"
 	"io"
 	"io/fs"
 	"os"
 	"os/exec"
+	"path/filepath"
 
 	"github.com/bitrise-io/go-utils/v2/command"
 	"github.com/bitrise-io/go-utils/v2/env"
 	"github.com/bitrise-io/go-utils/v2/log"
+	gozstd "github.com/klauspost/compress/zstd"
 )
 
+// ArchiveDependencyChecker ...
+//
+//go:generate moq -stub -out archive_dependency_checker_mock.go . ArchiveDependencyChecker
+type ArchiveDependencyChecker interface {
+	CheckDependencies() bool
+}
+
+// DependencyChecker ...
+type DependencyChecker struct {
+	logger  log.Logger
+	envRepo env.Repository
+}
+
+// NewDependencyChecker ...
+func NewDependencyChecker(logger log.Logger, envRepo env.Repository) *DependencyChecker {
+	return &DependencyChecker{
+		logger:  logger,
+		envRepo: envRepo,
+	}
+}
+
+// CheckDependencies ...
+func (dc *DependencyChecker) CheckDependencies() bool {
+	return dc.checkDepdendency("tar") && dc.checkDepdendency("zstd")
+}
+
+func (dc *DependencyChecker) checkDepdendency(binaryName string) bool {
+	cmdFactory := command.NewFactory(dc.envRepo)
+	cmd := cmdFactory.Create("which", []string{binaryName}, nil)
+	dc.logger.Debugf("$ %s", cmd.PrintableCommandArgs())
+
+	_, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		dc.logger.Warnf("%s is not present in $PATH, falling back to native implementation: %w", binaryName, err)
```

Comment:
I think we can delete this, in line 81 there is another message printed (and I prefer the info level instead of warning)